pulp_container/app/registry_api.py

```diff
@@ -1017,23 +1039,21 @@ def handle_safe_method(self, request, path, pk):
                 manifest = models.Manifest.objects.get(digest=pk, pk__in=repository_version.content)
             except models.Manifest.DoesNotExist:
                 repository = repository.cast()
-                if repository.PUSH_ENABLED:
-                    # the manifest might be a part of listed manifests currently being uploaded
-                    try:
-                        manifest = repository.pending_manifests.get(digest=pk)
-                        manifest.touch()
-                    except models.Manifest.DoesNotExist:
-                        raise ManifestNotFound(reference=pk)
-                else:
-                    if distribution.remote:
-                        remote = distribution.remote.cast()
-                        manifest = self.fetch_manifest(remote, repository_version, repository, pk)
-                        if manifest is None:
-                            return redirects.redirect_to_content_app("manifests", pk)
+                # the manifest might be a part of listed manifests currently being uploaded
+                # or saved during the pull-through caching
+                try:
+                    manifest = repository.pending_manifests.get(digest=pk)
+                    manifest.touch()
+                except models.Manifest.DoesNotExist:
+                    pass
```

Comment:
why not raising the error?

Reply:
There is still a chance that the fired pull-through download task has not finished and the user is trying to get a new listed manifest. We do not pre-record all listed manifests and their blobs in content-app. So, the manifest does not exist in `pending_manifests` and is still not associated with any repository.